### ISSUE TITLE
Float format in `checksum` validation

### ISSUE THREAD

**Issue by vikulikov** [fabianboesiger/ftx#120]
I believe that this comment from ftx documentation hasn't been taken into account while checksum validation has been implementing

```
Rules for float formatting for the checksum:

    Values smaller than 1e-04 (0.0001) should be formatted using scientific notation, and should contain zeroes before the exponent. For example, a message containing 7.5e-5 or 0.000075 should be formatted like so for computing the checksum: '7.5e-05'.
```

**Comment by MaxFangX** [comment]
Ahh yup, that is indeed missing in the checksum implementation. Looks like they finally added some docs explaining how the decimal / float formatting is supposed to work - it was really frustrating having to reverse engineer an implementation initially. Are you interested in tackling this issue?

**Comment by vikulikov** [comment]
Yes, I'm interested in fixing this. But I have no idea how to implement it efficiently.
For now I've come up with this implementation:
```rust
   fn format_value(&self, value: &Decimal) -> String {
        if value.fract() == dec!(0) {
            format!("{value:.1}")
        } else if *value < dec!(0.0001) {
            let mut formated = format!("{value:e}");
            let minus_idx = formated.find('-').expect("Passed abs(value) higher than 1");
            formated.insert(minus_idx + 1, '0');
            formated
        } else {
            value.to_string()
        }
```
It requires extra string allocation and linear search for '-' sign

**Comment by MaxFangX** [comment]
Looks good enough to me. I did some searching and chanced upon [this thread](https://internals.rust-lang.org/t/format-of-the-exponent-in-the-scientific-notation/9431) similarly asking how to pad the exponent with a 0 using the standard formatter - main point was that it might be possible to do this with some external crate, but for something so minor I don't think it's worth adding another dependency.

The `ws::Ws::_pad_0` function to be replaced already does a string allocation, so this implementation has essentially the same performance.

**Comment by MaxFangX** [comment]
Fixed by #129
